x/supplier/keeper/msg_server_create_claim.go

```diff
@@ -88,6 +54,9 @@ func (k msgServer) CreateClaim(goCtx context.Context, msg *suppliertypes.MsgCrea
 		SessionHeader:   msg.GetSessionHeader(),
 		RootHash:        msg.RootHash,
 	}
+
+	// TODO_TECHDEBT: check if this claim already exists and return an appropriate error
```

Comment:
This endangers the integrity of the blocker, so it's a blocker.

```suggestion
	// TODO_BLOCKER: check if this claim already exists and return an appropriate error
```